GetStream/stream-chat-android
# [4616] [V6] Port `chatclient.getmessage` over to `OfflinePlugin`

### 🎯 Goal

closes #4616 

### 🛠 Implementation details

Introduced a listener called `GetMessageListener` and it's database implementation. This listener is used to store the fetched message into the database  immediately once the `ChatClient.getMessage()` method is called.

### 🧪 Testing

<details>

<summary>Patch which disables sync</summary>

```
Index: stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/sync/messages/internal/SyncMessagesWork.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/sync/messages/internal/SyncMessagesWork.kt b/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/sync/messages/internal/SyncMessagesWork.kt
--- a/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/sync/messages/internal/SyncMessagesWork.kt	(revision 5fa983fb24eec038507f454a6103f7fcf663f711)
+++ b/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/sync/messages/internal/SyncMessagesWork.kt	(date 1673866797744)
@@ -53,7 +53,7 @@
             val syncManager = client.resolveDependency<StateAwarePlugin, SyncHistoryManager>() ?: error(
                 "No SyncHistoryManager found in StatePlugin."
             )
-            syncManager.sync()
+            // syncManager.sync()
 
             Result.success()
         } catch (e: Throwable) {
```

</details>

The testing sections aims to confirm that the newly implemented OP port indeed works and stores the message inside the database. In order to do that we disable sync since having sync turned on might give us false positives.

1. Open the app and log in with a user
2. Kill the app
3. Send a message from another device so that you get a notification
4. Turn off network connection on your device
5. open the message from the push notification
6. Open the database inspector and make and make an SQL query for the new message that you've received

Expected: The message should not be present in the database in the `v5` branch, but should be present in the app run from the branch in this PR.  Also, you will probably not see the message visually inside the app on `v5`, but will on this branch.


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media3.giphy.com/media/RCDeoJpz6zYglRxvQM/giphy.gif?cid=ecf05e473kin1pt3vr0ui740v427uqc0ccdntdsmfpgocncp&rid=giphy.gif&ct=g)
